tests/ui/unusable_matches_binding.stderr

```diff
@@ -0,0 +1,42 @@
+error: using identificator (`unusable_binding`) as matches! pattern argument without guard matches all cases
+  --> $DIR/unusable_matches_binding.rs:18:44
+   |
+LL |     let _ = matches!(matching_data_source, unusable_binding);
+   |                                            ^^^^^^^^^^^^^^^^
+   |
+note: matches! invocation always evaluates to `true`
+  --> $DIR/unusable_matches_binding.rs:18:13
+   |
+LL |     let _ = matches!(matching_data_source, unusable_binding);
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   = note: `-D clippy::unusable-matches-binding` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(clippy::unusable_matches_binding)]`
+   = note: this error originates in the macro `matches` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: using identificator (`used_binding`) as matches! pattern argument matches all cases
```

Comment:
The error message is pretty confusing for this one. The argument here won't match all cases.

The only special case is probably some mutating expression in `matches!` macro. But, I don't think this is a common case, and shouldn't be written that way anyway. We should only care about this when we are writing clippy suggestions.

Simplifying the logic to just check unused binding should be good enough for this lint.

Reply:
After the PR change we need also to check the case when the guard is present because in that situation different error message should be displayed (compared to the case when the guard is missing).